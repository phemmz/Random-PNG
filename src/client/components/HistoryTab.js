import React from 'react';

import {
  TabDetails
} from './Shared';

const HistoryTab = ({
  filenames, getFileDetails, generatedNumberData
}) => {
  return (
    <div className="hometab--container">
      <div className="tab--content">
        <div className="tab--content--left">
          <span className="history--list--title">Previously Generated Numbers</span>
          <ul className="history--list--container">
            {filenames.length ?
              filenames.map(filename =>
                <li
                  key={filename}
                  onClick={() => getFileDetails(filename)}
                >{filename}</li>
              ) : null
            }
          </ul>
        </div>
        <div className="tab--content--right">
          <div className="generated--title--container">
            <span className="history--list--title">{generatedNumberData.generatedPhoneNumberId}</span>
          </div>
          <TabDetails
            generatedNumberData={generatedNumberData}
          />
        </div>
      </div>
    </div>
  );
}

export default HistoryTab;
