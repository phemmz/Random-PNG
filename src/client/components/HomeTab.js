import React from 'react';
import classnames from 'classnames';

import {
  PngButton,
  TabDetails
} from './Shared';

const HomeTab = ({
  numberGenerated, handleInputChange, generateRandomPhoneNumbers,
  totalAmountToGenerate, onClick, isGenerating, errorMessage, generatedNumberData
}) => {
  return (
    <div className="hometab--container">
      {numberGenerated ?
        <div className="tab--content">
          <div className="tab--content--left">
            <span className="tab--input--amount--label">Amount</span>
            <div className="tab--input--container">
              <input
                name="totalAmountToGenerate"
                type="number"
                className="tab--input"
                onChange={handleInputChange}
                value={totalAmountToGenerate}
              />
            </div>
            <PngButton
              classNames="w-100"
              btnText={isGenerating ? "Generating..." : "Generate"}
              loading={isGenerating}
              handleClick={generateRandomPhoneNumbers}
            />
            {errorMessage ? <span className="error--message">{errorMessage}</span> : null}
          </div>
          <div className="tab--content--right">
            <TabDetails generatedNumberData={generatedNumberData} />
          </div>
        </div> :
        <div className={classnames("tab--content", "tab--content--active")}>
          <span className="tab--title"><span className="tab--title--bold">Random-PNG</span> is the app that generates random phone numbers for you. Each number is unique and has a 10 digit length.</span>
          <PngButton
            btnText="Generate Phone Numbers"
            handleClick={onClick}
          />
        </div>
      }
    </div>
  );
}

export default HomeTab;
