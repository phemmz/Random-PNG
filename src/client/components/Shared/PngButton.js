import React from 'react';

const PngButton = ({ handleClick, classNames, btnText, loading = false }) => {
  return (
    <button
      className={`generate--numbers--btn ${classNames}`}
      onClick={handleClick}
      disabled={loading}
    >{btnText}</button>
  );
}

export default PngButton;
