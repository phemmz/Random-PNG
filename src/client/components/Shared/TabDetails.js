import React from 'react';
import classnames from 'classnames';

const TabDetails = ({
  generatedNumberData: {
    totalPhoneNumbersGenerated, min, max, generatedPhoneNumbers = []
  }
}) => {
  return (
    <div className="details--container">
      <div className="tab--details--section">
        <div className="section--info">
          <span className="details--label">Total</span>
          <div className="tab--details--container">{totalPhoneNumbersGenerated}</div>
        </div>
        <div className="section--info">
          <span className="details--label">Min number</span>
          <div className="tab--details--container">{min}</div>
        </div>
        <div className="section--info">
          <span className="details--label">Max number</span>
          <div className="tab--details--container">{max}</div>
        </div>
      </div>
      <div className={classnames("section--info", "tab--list--container")}>
        <span className="details--label">Numbers generated</span>
        <div className={classnames("tab--details--container", "h-100")}>
          {generatedPhoneNumbers.length ?
            generatedPhoneNumbers.map(generatedPhoneNumber =>
              <div
                key={generatedPhoneNumber}
                className="db"
              >
                {generatedPhoneNumber}
              </div>) :
            null
          }
        </div>
      </div>
    </div>
  );
}

export default TabDetails;
