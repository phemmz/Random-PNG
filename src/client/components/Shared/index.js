import PngButton from './PngButton';
import TabDetails from './TabDetails';

export {
  PngButton,
  TabDetails
};
