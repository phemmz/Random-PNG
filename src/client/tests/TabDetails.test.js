import React from "react";
import { shallow } from "enzyme";

import { TabDetails } from "../components/Shared";

describe("TabDetails", () => {
  const props = {
    generatedNumberData: {
      generatedPhoneNumberId: 'number-1560053355220',
      generatedPhoneNumbers: ['0101507997', '0999875346'],
      min: '0101507997',
      max: '0999875346',
      totalPhoneNumbersGenerated: 1000
    }
  };
  const wrapper = shallow(<TabDetails {...props} />);

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
