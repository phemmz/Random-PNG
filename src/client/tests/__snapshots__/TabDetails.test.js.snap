// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabDetails should render correctly 1`] = `ShallowWrapper {}`;
